docs/drone.adoc

```diff
@@ -0,0 +1,110 @@
+
+You can read more about Drone in https://docs.jboss.org/author/display/ARQ/Drone
+
+=== Integration with Cube Docker
+
+Arquillian Cube Docker is the Arquillian extension for managing Docker containers from your test.
+
+Selenium offers Docker images for Selenium Standalone Server with Chrome or Firefox installed.
+You can see the images at https://github.com/SeleniumHQ/docker-selenium.
+
+With these images and using Selenium WebDriver you can run your functional tests for web IU applications without having to install any browser.
+
+It seems logic an integration of Arquillian Drone with Cube, so you can use all powerful features of Drone, and let Cube configure Docker things for you.
+
+This integration does next things for you:
+
+. Inspect classpath to get selenium version used
+. Starts a docker container with configured browser in Drone and same selenium version as your `JARs`. If not Firefox is used
+. Provides a WebDriver that is able to connect to this container
+. If configured (by default it is) a VNC recorder container is started so each test is recorded in a `flv` file.
+
+=== Configuration
+
+Apart from using Drone configuration properties for configuring browser, Cube Docker can be customized with some specific attributes:
+
+recordingMode:: Configures mode for recording. The valid values are: `ALL, ONLY_FAILING, NONE. Default value is `ALL`.
+videoOutput:: Directory where video is stored. By default is `target/surefire-target` if not exists `build/test-results` if not `build` and if not `target`.
+
+An example of configuration might look like:
+
+[source, xml]
+.arquillian.xml
+----
+<arquillian>
+    <extension qualifier="cubedrone">
+        <property name="recordingMode">NONE</property>
+</arquillian>
+----
+
+=== Example
+
+For adding Drone Cube integration you only need a Cube project with a `docker-compose` or `cube` file only specifying business containers.
+
+[source, yml]
+----
+helloworld:
+  image: dockercloud/hello-world
+  ports:
+    - "80:80"
+----
+
+Configure the Cube extension:
+
+[source, xml]
+.arquillian.xml
+----
+<extension qualifier="docker">
+    <property name="definitionFormat">COMPOSE</property>
+    <property name="machineName">dev</property>
+    <property name="dockerContainersFile">docker-compose.yml</property>
+</extension>
+----
+
+And finally among `cube` dependencies, add the `drone`, `selenium` and `cube-drone` dependencies.
+
+[source, xml]
+.pom.xml
+----
+<dependencyManagement>
+    <dependencies>
+        <dependency>
+            <groupId>org.jboss.arquillian.extension</groupId>
+            <artifactId>arquillian-drone-bom</artifactId>
+            <version>2.0.0.Final</versio>
+            <type>pom</type>
+            <scope>import</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.arquillian.selenium</groupId>
+            <artifactId>selenium-bom</artifactId>
+            <version>2.53.1</version>
```

Comment:
Yes true, I have not explained correctly, but checking all the docs it uses
this approach. I think it is the correct one to follow in our examples too,
isn't it?

El dt., 2 ag. 2016 a les 15:54, Matous Jobanek (<notifications@github.com>)
va escriure:

> In docs/drone.adoc
> <https://github.com/arquillian/arquillian-cube/pull/402#discussion_r73158634>
> :
>
> > +[source, xml]
> > +.pom.xml
> > +----
> > +<dependencyManagement>
> > +    <dependencies>
> > +        <dependency>
> > +            <groupId>org.jboss.arquillian.extension</groupId>
> > +            <artifactId>arquillian-drone-bom</artifactId>
> > +            <version>2.0.0.Final</versio>
> > +            <type>pom</type>
> > +            <scope>import</scope>
> > +        </dependency>
> > +        <dependency>
> > +            <groupId>org.jboss.arquillian.selenium</groupId>
> > +            <artifactId>selenium-bom</artifactId>
> > +            <version>2.53.1</version>
>
> Well, not exactly. Arquillian-drone-bom manages a selenium version as
> well:
> https://repo1.maven.org/maven2/org/jboss/arquillian/extension/arquillian-drone-bom/2.0.0.Final/arquillian-drone-bom-2.0.0.Final.pom
> The version defined in this BOM is that one the Drone should be compatible
> with.
> The reason why the selenium-bom was created is that the Selenium's release
> cadence is much higher then the Drone's one. So, it can happen that the
> newest version of some browser is not supported by selenium version defined
> in Drone, then using selenium-bom the developer can easily specify selenium
> version that he really needs.
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/arquillian/arquillian-cube/pull/402/files/006d8aabc4c57f416401e739269bc028ca0e1c25#r73158634>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABcmYdQ_gdZm5tqD3WnjxQLYA6oojJHXks5qb0v4gaJpZM4JV7F4>
> .
>